jax/_src/tree.py

```diff
@@ -284,3 +286,45 @@ def unflatten(treedef: tree_util.PyTreeDef,
     - :func:`jax.tree.structure`
   """
   return tree_util.tree_unflatten(treedef, leaves)
+
+
+def field(*, static: bool = False, metadata: dict[str, Any] | None = None, **kwds) -> dataclasses.Field:
+  """Used to mark a jax.dataclass field as static.
+
+  See {func}`jax.tree.dataclass` for details.
+  """
+  return dataclasses.field(metadata=(metadata or {}) | {'static': static}, **kwds)
+
+
+@dataclass_transform(field_specifiers=(field, dataclasses.Field), frozen_default=True)
+def dataclass(cls: T, /, **kwargs) -> T:
+  """Create a pytree-compatible dataclass.
```

Comment:
Is the fact that we return a standard library dataclass part of our public API contract? E.g. can users assume that `dataclasses.replace` just works and `dataclasses.is_dataclass` returns True?

Reply:
It does with this implementation – I think that yes, that should be tested and be part of the contract. What do you think?